### ISSUE TITLE
Daq Viewer "initialize detector" crashes viewer / dashboard

### ISSUE THREAD

**Issue by bmoneke** [PyMoDAQ/PyMoDAQ#168]
Clicking on "initialize detector" of the 2d or 0d viewer crashes the viewer (and dashboard).

There is no error message in the log, but in the console is written something like "QThread: Destroyed while thread is still running". 

Viewer are the mock viewers or an application of the generic camera viewer.
Pymodaq version is 4.0.8
Pyqtgraph is the most recent version
Python 3.11
Windows 10

How to reproduce:
1. Start the dashboard in cli
2. Create a preset with a mockdetector
3. Load preset
4. Click on initialize detector

**Comment by seb5g** [comment]
In fact, the preset has already initialized the control modules (see the check box for that in the presets). It means that in fact you are uninitializing the modules by clicking on them. However this is true that there is an issue related to the thread when deinitializing.

**Comment by seb5g** [comment]
if you know a proper way to stop a thread, i'd like to hear it !!
So far I do:

```
try:
    self._hardware_thread.quit()
    self._hardware_thread.wait()
    finished = self._hardware_thread.isFinished()
    if finished:
        pass
    else:
        print('Thread still running')
        self._hardware_thread.terminate()
        self.update_status('thread is locked?!', 'log')
except Exception as e:
    self.logger.exception(str(e))
```

while the thread is initialized as:

```
hardware = DAQ_Detector(self._title, self.settings, self.detector)
self._hardware_thread = QThread()
if config('viewer', 'viewer_in_thread'):
    hardware.moveToThread(self._hardware_thread)

self._hardware_thread.hardware = hardware
self._hardware_thread.start()
```

**Comment by bmoneke** [comment]
I narrowed it down:
For the Move (which is working), the sequence at preset loading is: "Move_ui.send_init" and then "Move.init_hardware".
For the Viewer (which is not working) the "Viewer_ui.send_init" is never called, just "Viewer.init_hardware".

If I load another preset (a mock mover and a mock camera) I reproduce above results. If I do not initialize the detectors at preset loading, I can connect/disconnect at will.

If I do not initialize the camera in the beam_steering_mock preset at loading time, the preset never loads. It hangs at the first move (the first piezo) after the camera.

Conclusion: Somewhere the preset loading of viewers has a bug regarding viewer initialization.

**Comment by seb5g** [comment]
this should be solved by PR #199 into pymodaq_dev

**Comment by bmoneke** [comment]
Should have been closed by #174
